src/content/docs/en/reference/dev-overlay-reference.mdx

```diff
@@ -0,0 +1,260 @@
+  icon: '<svg>...</svg>',
+}
+```
+
+### `init: (canvas: ShadowRoot, eventTarget: EventTarget) => void`
+
+This is the core part of the plugin. This function will be called when the plugin is loaded, which will either be when the browser is idle or when the user clicks on the plugin in the UI.
+
+The function receives two arguments:
+
+#### `canvas`
+
+A ShadowRoot that the plugin can use to render its UI. Every plugin receive its own dedicated ShadowRoot for rendering its UI. Additionally, the parent element is positioned using `position: absolute;` and as such, the plugin UI automatically won't affect the layout of the page.
+
+```ts
+export default {
+	id: 'super-plugin',
+	name: 'My Super Plugin',
+	icon: '<svg>...</svg>',
+	init(canvas) {
+    canvas.appendChild(document.createTextNode('Hello World!'))
+  }
+}
+```
+
+#### `eventTarget`
+
+An [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) that can be used to send and receive events from the dev overlay.
+
+## Client-side Events
+
+Using the `eventTarget` argument on the `init` hook, plugins can send and receive events from the dev overlay. The following events are available:
+
+### `plugin-toggle`
+
+This event is fired when the user clicks on the plugin icon in the UI. The new state of the plugin is included in the event in the `detail` property.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.addEventListener('plugin-toggle', (event) => {
+        if (event.detail.state === true) {
+          console.log("The plugin is now enabled!")
+        }
+      })
+    }
+  }
+```
+
+### `plugin-notification`
+
+This event can be sent to inform the user that the plugin requires attention.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.dispatchEvent(
+          new CustomEvent('plugin-notification', {
+            detail: {
+              state: true,
+            },
+          })
+        );
+    }
+  }
+```
+
+`state` can be either `true` or `false`. When `true`, the plugin icon will be highlighted using a red dot. Conversely, when `false`, the highlight will be removed.
+
+## Client-Server Communication
+
+Using [Vite's methods for client-server communication](https://vitejs.dev/guide/api-plugin.html#client-server-communication), dev overlay plugins can communicate with the server.
+
+In addition to being able to send and receive arbritary messages, the dev overlay also sends the following messages, where `PLUGIN_ID` is the ID of a plugin:
+
+### `astro-dev-overlay:PLUGIN_ID:init`
+
+This message is sent when the plugin is initialized. The data for this message is empty.
+
+```ts title="integration.ts"
+
+```
+
+### `astro-dev-overlay:PLUGIN_ID:toggle`
+
+This message is sent when the user clicks on the plugin icon in the UI. The data for this message is a boolean indicating whether the plugin is enabled or not.
```

Comment:
```suggestion
This message is sent when the user clicks on the plugin icon in the UI. The data for this message is [the plugin's state](#state-boolean) indicating whether the plugin is enabled or not.
```
Would this be correct? This might tie it back nicely to the property!

Also, if we're showing an empty message above, is it worth showing the message being either true or false here, too?